src/app/components/Table/useClientSidePagination.ts

```diff
@@ -0,0 +1,102 @@
+import { To, useSearchParams } from 'react-router-dom'
+import { AppErrors } from '../../../types/errors'
+import { ComprehensivePaginationEngine } from './PaginationEngine'
+import { List } from '../../../oasis-nexus/api'
+import { TablePaginationProps } from './TablePagination'
+
+type ClientSizePaginationParams<Item> = {
+  paramName: string
+  clientPageSize: number
+  serverPageSize: number
+  filter?: (item: Item) => boolean
+}
+
+const knownListKeys: string[] = ['total_count', 'is_total_count_clipped']
+
+function findListIn<T extends List, Item>(data: T): Item[] {
+  const candidates = Object.keys(data).filter(
+    key => !knownListKeys.includes(key) && Array.isArray(data[key as keyof T]),
+  ) as (keyof T)[]
+  switch (candidates.length) {
+    case 0:
+      throw new Error('Data not found!')
+    case 1:
```

Comment:
With strict types this could be ensured at compile time, but not worth inventing it for this temporary workaround until backend adds filtering by signature